src/index.ts

```diff
@@ -1,220 +1,143 @@
-import { setImplementations } from "webnative/setup.js"
+import * as Session from "webnative/session"
+import * as Webnative from "webnative"
 
-import { getSimpleLinks } from "webnative/fs/protocol/basic.js"
-import { PublicFile } from "webnative/fs/v1/PublicFile.js"
-import { PublicTree } from "webnative/fs/v1/PublicTree.js"
-import FileSystem from "webnative/fs/filesystem.js"
+import * as BaseCrypto from "./components/crypto/implementation/base.js"
+import * as BaseManners from "./components/manners/implementation/base.js"
+import * as EthereumWallet from "./wallet/implementation/ethereum.js"
+import * as Wallet from "./wallet/implementation.js"
 
-import * as readKey from "./read-key"
-import * as walletUcan from "./ucan"
-import * as wallet from "./wallet"
+import { did } from "./wallet/common.js"
 
-import { USE_WALLET_AUTH_IMPLEMENTATION } from "./auth/implementation"
-import { hasProp } from "./common"
 
 
+// 🛳
 
-// ⛰
 
+export const components = {
+  crypto(settings: Configuration & {
+    wallet: Wallet.Implementation
+  }): Promise<Crypto.Implementation> {
+    return BaseCrypto.implementation(settings.wallet, {
+      storeName: namespace(settings),
+      exchangeKeyName: "exchange-key",
+      writeKeyName: "write-key"
+    })
+  },
 
-export const ROOT_PERMISSIONS = { fs: { private: [ path.root() ], public: [ path.root() ] } }
-export const READ_KEY_PATH = path.file(path.Branch.Public, ".well-known", "read-key")
-
-
-
-// 🌳
-
-
-export type AppOptions = {
-  onAccountChange?: (appState: AppState) => unknown
-  onDisconnect?: (...args: unknown[]) => unknown
-  resetWnfs?: boolean
-  useWnfs?: boolean
+  manners(settings: Configuration & {
+    crypto: Crypto.Implementation
+    wallet: Wallet.Implementation
+  }): Promise<Manners.Implementation> {
+    return BaseManners.implementation(
+      settings.wallet,
+      { configuration: settings }
+    )
+  }
 }
 
 
 
 // 🚀
 
 
-export async function app(options?: AppOptions): Promise<AppState> {
-  let fs
+export type Options = {
+  onAccountChange?: (program: Program) => unknown
+  onDisconnect?: () => unknown
 
-  options = options || {}
-
-  const { resetWnfs = false, useWnfs = true } = options
+  wallet?: Wallet.Implementation
+}
 
-  setImplementations(USE_WALLET_AUTH_IMPLEMENTATION)
 
-  // Check if browser is supported
-  if (hasProp(self, "isSecureContext") && self.isSecureContext === false) throw InitialisationError.InsecureContext
-  if (await isSupported() === false) throw InitialisationError.UnsupportedBrowser
+/**
+ * 🚀 Build a webnative-walletauth program.
+ *
+ * Contrary to a regular webnative program,
+ * this'll create an account for you automatically.
+ * It also does session management for you.
+ *
+ * You may pass in a custom `wallet` implementation
+ * if you'd like to use another wallet than the built-in
+ * Ethereum one.
+ *
+ * See webnative's `program` function documentation for more info.
```

Comment:
Might be useful to provide a link to the `program` documentation, though i guess we may need to wait for the new guide to be published first?

Reply:
Added a link, which won't work currently, but will once 0.35 docs are published.